instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/server/RemoteServerInstrumentation.java

```diff
@@ -28,7 +29,8 @@
 public class RemoteServerInstrumentation implements TypeInstrumentation {
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return implementsInterface(named("java.rmi.Remote"));
+    return implementsInterface(named("java.rmi.Remote"))
+        .and(not(nameStartsWith("org.springframework.remoting")));
```

Comment:
This may seem like a somewhat stupid question, but what's the reason for a separate spring-rmi instrumentations? Did it not work properly before? It looks like you have to exclude the spring RMI classes from the "standard" Java RMI instrumentation.

Reply:
The standard Java RMI instrumentation didn't work properly for Spring RMI.
Client side spans don't get generated and in Server side span it instruments an Spring RMI internal class: `RmiInvocationHandler` because it extends `java.rmi.Remote`, which wasn't the original intent in the Java RMI instrumentation as there the user defined class extends it. Due to this it captures `rpc.service` and `rpc.method` incorrectly.
And to suppress this unwanted instrumentation I had to include that condition in Java RMI instrumentation.

Reply:
Got it, thanks for the explanation!